### PR TITLE
100 flexible time

### PR DESCRIPTION
**Pull request by Kafcis**
This one is quite a big one because of modifications to the way Task works. Almost all the tests broke fixed them by assigning all the flexibility to false, made some new ones as well. 

The Storage ,AddCommand ,Task and Parser Classes was reworked to accommodate the additional parameter. Not all the code is new, it has 95 the time allocate skeleton and 71 the comma workaround inside it as well. 

At the moment Task flexibility is kept in the Task Class, it should be moved to Time class during the refactoring of all the time related attributes of Task

### CODE REVIEW

**Review by amanda-chua — APPROVED** [review]
Wow that's a lot of change but looks good!

File: src/main/java/athena/Parser.java [hunk 1]
```diff
@@ -76,11 +76,12 @@ public static String getParameterDesc(String taskInformation, String delimiter,
      */
     public static Command parseAddCommand(String taskInfo, int namePos, int timePos, int durationPos, int deadlinePos,
                                           int recurrencePos, int importancePos, int addNotesPos)
-                                          throws InvalidCommandException {
+            throws InvalidCommandException {
         String nullDefault = "";
         String name = getParameterDesc(taskInfo, NAME_DELIMITER, namePos, nullDefault);
         //TODO: allow for empty string, assign flexible attribute, true if string is null, false if filled
         String time = getParameterDesc(taskInfo, TIME_DELIMITER, timePos, nullDefault);
+        boolean flexible = (time == nullDefault);
```
Comment (daniellimws) [inline src/main/java/athena/Parser.java]: `isFlexible` according to coding standards

File: src/main/java/athena/TaskList.java [hunk 2]
```diff
@@ -57,8 +57,8 @@ public TaskList(ArrayList<Task> tasks) {
      * @return Task as Task object
      */
     private Task createTask(int number, String name, String startTime, String duration, String deadline,
-                            String recurrence, Importance importance, String notes) {
-        Task task = new Task(name, startTime, duration, deadline, recurrence, importance, notes, number);
+                            String recurrence, Importance importance, String notes, Boolean flexible) {
```
Comment (daniellimws) [inline src/main/java/athena/TaskList.java]: `isFlexible`

File: src/main/java/athena/TaskList.java [hunk 3]
```diff
@@ -97,11 +97,11 @@ public void addTask(Task task) {
      */
 
     public void addTask(int number, String name, String startTime, String duration,
-                        String deadline, String recurrence, Importance importance, String notes) {
+                        String deadline, String recurrence, Importance importance, String notes, boolean flexible) {
```
Comment (daniellimws) [inline src/main/java/athena/TaskList.java]: `isFlexible`

File: src/main/java/athena/TaskList.java [hunk 4]
```diff
@@ -117,9 +117,9 @@ public void addTask(int number, String name, String startTime, String duration,
      * @param notes      Additional notes of task
      */
     public void addTask(String name, String startTime, String duration,
-                        String deadline, String recurrence, Importance importance, String notes) {
+                        String deadline, String recurrence, Importance importance, String notes, Boolean flexible) {
```
Comment (daniellimws) [inline src/main/java/athena/TaskList.java]: `isFlexible`

File: src/main/java/athena/commands/AddCommand.java [hunk 5]
```diff
@@ -17,6 +18,7 @@
     private String taskRecurrence;
     private Importance taskImportance;
     private String taskNotes;
+    private Boolean taskFlexible;
```
Comment (daniellimws) [inline src/main/java/athena/commands/AddCommand.java]: `isTaskFlexible`?

File: src/main/java/athena/commands/AddCommand.java [hunk 6]
```diff
@@ -28,9 +30,10 @@
      * @param recurrence String representing recurrence of task.
      * @param importance String representing importance of task.
      * @param notes      String representing additional notes of task.
+     * @param flexible  Boolean representing if task time is flexible
      */
     public AddCommand(String name, String startTime, String duration, String deadline,
-                      String recurrence, String importance, String notes) {
+                      String recurrence, String importance, String notes, boolean flexible) {
```
Comment (daniellimws) [inline src/main/java/athena/commands/AddCommand.java]: `isFlexible`?

File: src/main/java/athena/task/Task.java [hunk 7]
```diff
@@ -23,6 +22,7 @@
 
     private String recurrence;
     private LocalDate recurrenceDate;
+    private boolean flexible;
```
Comment (daniellimws) [inline src/main/java/athena/task/Task.java]: `isFlexible`? The program still works when there is a method called `isFlexible`

File: src/main/java/athena/task/Task.java [hunk 8]
```diff
@@ -44,18 +44,18 @@ private String getStatus() {
...
      * @param deadline   when the task is due
      * @param recurrence when the task repeats
      * @param importance importance of the task
      * @param notes      additional notes for the task
      * @param number     task number
+     * @param flexible  time flexibility
      */
     public Task(String name, String startTime, String duration, String deadline,
-                String recurrence, Importance importance, String notes, int number) {
+                String recurrence, Importance importance, String notes, int number, Boolean flexible) {
```
Comment (daniellimws) [inline src/main/java/athena/task/Task.java]: `isFlexible`?

**Review by daniellimws — COMMENTED** [review]
Looks good. Just one thing, I think `flexible` as the name for a `Boolean` might violate the coding standards. I think something like `isFlexible` or `flexibility` would be better.

**Pull request merged**